src/chains-config/cache/QueryFeeDetailsCache.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * When checking the cache there are 3 possible choices to return.
+ */
+enum QueryFee {
+	/**
+	 * queryFeeDetails is available
+	 */
+	available = 'available',
+	/**
+	 * queryFeeDetails is not available
+	 */
+	notAvailable = 'notAvailable',
+	/**
+	 * We dont know is queryFeeDetails is available
+	 */
+	unknown = 'unknown',
+}
+
+export class QueryFeeDetailsCache {
+	/**
+	 * Highest known runtime that doesn't have queryFeeDetails.
+	 */
+	private _versionWithoutCall;
+	/**
+	 * Lowest known runtime that has queryFeeDetails
+	 */
+	private _versionWithCall;
+
+	constructor(
+		versionWithoutCall: number | null,
+		versionWithCall: number | null
+	) {
+		this._versionWithoutCall = versionWithoutCall;
+		this._versionWithCall = versionWithCall;
+	}
+
+	public isQueryFeeDetailsAvail(specVersion: number) {
+		const { available, notAvailable, unknown } = QueryFee;
+
+		if (
+			this._versionWithCall !== null &&
+			specVersion >= this._versionWithCall
+		) {
+			return available;
+		}
+
+		if (
+			this._versionWithoutCall !== null &&
+			specVersion <= this._versionWithoutCall
+		) {
+			return notAvailable;
+		}
+
+		return unknown;
+	}
+
+	public setVersionWithoutCall(specVersion: number) {
```

Comment:
I'm not sure that using `set` here is a good idea, as it may not be sent if a lower version already exist.

So perhaps `registerVersionWithoutApi`or something.

Reply:
I think the `registerWithoutApi` sounds great. Getting the naming right with these things can feel tough sometimes, but this is a good fit.

Reply:
yupp, naming is hard :)

Reply:
I do think we should keep the `set` though after doing some refactoring, as it will keep naming differences between the classes instance variables seperated, but that being said I will clarify what it does with docs.

Reply:
fixed